pkg/apis/v1alpha1/ippool_types.go

```diff
@@ -0,0 +1,90 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+// +kubebuilder:object:generate=true
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// IPPool is the Schema for the ippools API
+type IPPool struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   IPPoolSpec   `json:"spec"`
+	Status IPPoolStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// IPPoolList contains a list of IPPool
+type IPPoolList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []IPPool `json:"items"`
+}
+
+// IPPoolSpec defines the desired state of IPPool.
+type IPPoolSpec struct {
+	// Type defines the type of this IPPool, public or private.
+	Type string `json:"type"`
+	// Subnets defines set of subnets need to be allocated.
+	Subnets []SubnetRequest `json:"subnets"`
+}
+
+// IPPoolStatus defines the observed state of IPPool.
+type IPPoolStatus struct {
+	// Subnets defines subnets allocation result.
+	Subnets []SubnetResult `json:"subnets"`
+	// Conditions defines current state of the IPPool.
+	Conditions []IPPoolCondition `json:"conditions"`
+}
+
+// SubnetRequest defines the subnet allocation request.
+type SubnetRequest struct {
+	// PrefixLength defines prefix length for this subnet.
+	// +optional
+	PrefixLength int `json:"prefixLength,omitempty"`
+
+	// IPFamily defines the IP family type for this subnet, could be IPv4 or IPv6.
+	// This is optional, the default is IPv4.
```

Comment:
// +kubebuilder:default=IPv4 to define the default value?

Reply:
ok